addressbook/interface/exchangeAutoCompleteResult/mivExchangeAutoCompleteResult.js

```diff
@@ -169,36 +170,30 @@ mivExchangeAutoCompleteResult.prototype = {
                         emailList = emailList + ",";
                     }
                     var tmpCard = childNodes.getNext().QueryInterface(Ci.mivExchangeAbCard);
-                    //dump(" || "+tmpCard+"\n");
-                    //					emailList = emailList + tmpCard.displayName + " <" + tmpCard.primaryEmail + ">" ;
                     emailList = emailList + tmpCard.firstName + " " + tmpCard.lastName + " <" + tmpCard.primaryEmail + ">";
                 }
-                //dump("  ++ addr:"+emailList+"\n");
-                return emailList;
+                result = emailList;
             }
-            //dump("  -- mailListURI:"+this._cards[aIndex].mailListURI+"\n");
         }
-        //		return this._cards[aIndex].displayName + " <" + this._cards[aIndex].primaryEmail + ">";
-        //dump("firstname:"+this._cards[aIndex].firstName+", lastname:"+this._cards[aIndex].lastName+", displayName:"+this._cards[aIndex].displayName+"\n");
-        if ((this._cards[aIndex].firstName != "") || (this._cards[aIndex].lastName != "")) {
-            return this._cards[aIndex].firstName + " " + this._cards[aIndex].lastName + " <" + this._cards[aIndex].primaryEmail + ">";
+        else if (card.firstName != ""
+                || card.lastName != "") {
+            result = card.firstName + " " + card.lastName + " <" + card.primaryEmail + ">";
+        }
+        else if (card.displayName != "") {
```

Comment:
I would assume `displayName` should have higher priority than `firstName` + `lastName`.

Reply:
Well, I think you are right :+1: 

What will be the ideal output when we have all the informations ?

Something like : `DisplayName (FirstName LastName) <email@example.com>`, like, for example, `Trim (Adrien Dorsaz) <adrien@example.com>` ?

Reply:
Works for me.

Reply:
Well, I've tested that idea and I've discovered the text returned by this function is used directly as the "TO:" SMPT header. As almost every display name is a concatenation of first and last name, that looked wired.

Although, we have a comment part we can add on each search result which is not added to the "TO:" header (text after the ` ̶ ` char). I've decided to add the first name and last name there. So, for example, you'll see a result like: `Trim <adrien@example.com> ̶  Adrien Dorsaz, Exchange Calendar`.